node/node.go

```diff
@@ -756,6 +825,104 @@ func (node *AlgorandFullNode) GetTransactionByID(txid transactions.Txid, rnd bas
 	}, nil
 }
 
+// StartCatchup starts the catchpoint mode and attempt to get to the provided catchpoint
+// this function is intended to be called externally via the REST api interface.
+func (node *AlgorandFullNode) StartCatchup(catchpoint string) error {
+	node.mu.Lock()
+	defer node.mu.Unlock()
+	if node.indexer != nil {
+		return fmt.Errorf("catching up using a catchpoint is not supported on indexer-enabled nodes")
+	}
+	if node.catchpointCatchupService != nil {
+		stats := node.catchpointCatchupService.GetStatistics()
+		return fmt.Errorf("unable to start catchpoint catchup for '%s' - already catching up '%s'", catchpoint, stats.CatchpointLabel)
+	}
+	var err error
+	node.catchpointCatchupService, err = catchup.MakeNewCatchpointCatchupService(catchpoint, node, node.log, node.net, node.ledger.Ledger)
+	if err != nil {
+		node.log.Warnf("unable to create catchpoint catchup service : %v", err)
+		return err
+	}
+	node.catchpointCatchupService.Start(node.ctx)
+	return nil
+}
+
+// AbortCatchup aborts the given catchpoint
+// this function is intended to be called externally via the REST api interface.
+func (node *AlgorandFullNode) AbortCatchup(catchpoint string) error {
+	node.mu.Lock()
+	defer node.mu.Unlock()
+	if node.catchpointCatchupService == nil {
+		return nil
+	}
+	stats := node.catchpointCatchupService.GetStatistics()
+	if stats.CatchpointLabel != catchpoint {
+		return fmt.Errorf("unable to abort catchpoint catchup for '%s' - already catching up '%s'", catchpoint, stats.CatchpointLabel)
+	}
+	node.catchpointCatchupService.Abort()
+	return nil
+}
+
+// SetCatchpointCatchupMode change the node's operational mode from catchpoint catchup mode and back
+func (node *AlgorandFullNode) SetCatchpointCatchupMode(catchpointCatchupMode bool) (newCtx context.Context) {
```

Comment:
Is my understanding correct?
1. If prev catchupointing is detected in `MakeFull` then continue catching up
2. If one calls `StartCatchup` then `catchpointCatchupMode = true` and force all services to stop
3. When catchupointing is done then `catchpointCatchupMode = false` and start all the services on updated ledger.

Reply:
that sounds correct.